pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -74,6 +75,24 @@ public JavaInstance(ContextImpl contextImpl, Object userClassObject, InstanceCon
         }
     }
 
+    public void setup() {
+        if (null != function && function instanceof Hook) {
+            try {
+                ((Hook) function).preProcess(context);
+            } catch (RuntimeException e) {
+                log.error("setup error:", e);
+                throw new RuntimeException("function preProcess occurred exception", e);
+            }
+        }
+        if (null != javaUtilFunction && javaUtilFunction instanceof Hook) {
```

Comment:
according to the [docs](http://pulsar.apache.org/docs/en/next/functions-package/#java), `java.util.function.Function` is not intend to interact with `Context`, so we should avoid to passing `Context` to `java.util.function.Function`.

Reply:
I have corrected it